### PR TITLE
Allow the concatenation of independent data sets with common tracers

### LINKED ISSUE #62: Allow concatenation of two independent data sets with common tracers
I want to concatenate two different datasets that are statistically independent, but share have a common tracer (CMB lensing). At the moment, `concatenate_data_sets` does not allow this: https://github.com/LSSTDESC/sacc/blob/e4c23bcaa1d58862cb91f8543ce6b83569ab9d3c/sacc/sacc.py#L1173.

### PR DESCRIPTION
**Pull request by carlosggarcia**
Closes #62

### CODE REVIEW

File: sacc/sacc.py [hunk 1]
```diff
@@ -1293,9 +1305,11 @@ def concatenate_data_sets(*data_sets, labels=None):
 
             # Check for clashing metadata
             if key in output.metadata:
-                raise ValueError("Metadata in concatenated Saccs have "
-                                 "same name. Set the labels parameter "
-                                 "to fix this.")
-            output.metadata[key] = val
+                if key not in same_tracers:
```
Comment (damonge) [inline sacc/sacc.py:1308]: Why do you need this. Are the tracer names stored as metadata?

File: test/test_sacc2.py [hunk 2]
```diff
@@ -550,6 +567,7 @@ def test_concatenate_data():
         s3.get_tracer(t2)
 
 
+
```
Comment (damonge) [inline test/test_sacc2.py:570]: too many empty lines?

**Review by damonge — COMMENTED** [review]
I think it looks good. Just a few questions. @joezuntz may want to take a look too.

**Comment by carlosggarcia** [comment]
@damonge back to you. I have added the possibility of using both `labels` and `same_tracers`

File: sacc/sacc.py [hunk 3]
```diff
@@ -1190,6 +1190,11 @@ def concatenate_data_sets(*data_sets, labels=None):
         Optional list of strings to append to tracer and metadata names, in
         case of a clash.
 
+    same_tracers: List[str]
+        Optional list of tracers that are assumed to be the same in the
+        different data_sets but with no correlation between the data points in
+        each of them.
```
Comment (damonge) [inline sacc/sacc.py:1196]: I'd rephrase:
```
but with no correlation between the data points involving them.
```

File: test/test_sacc2.py [hunk 4]
```diff
@@ -549,6 +566,13 @@ def test_concatenate_data():
         s3.get_tracer(t1)
         s3.get_tracer(t2)
 
+    # labels + same_tracers
```
Comment (damonge) [inline test/test_sacc2.py:569]: good test

File: sacc/sacc.py (hunk 3 above)
Comment (carlosggarcia) [reply sacc/sacc.py]: Done

File: sacc/sacc.py [hunk 5]
```diff
@@ -1190,6 +1190,11 @@ def concatenate_data_sets(*data_sets, labels=None):
         Optional list of strings to append to tracer and metadata names, in
         case of a clash.
 
+    same_tracers: List[str]
+        Optional list of tracers that are assumed to be the same in the
+        different data_sets but with no correlation between the data points
+        involving them.
+
```
Comment (joezuntz) [inline sacc/sacc.py:1197]: Could you explain in theis docstring what wil happen to these data sets?
